controllers/kafkaacl_controller_test.go

```diff
@@ -71,11 +71,42 @@ var _ = Describe("Kafka ACL Controller", func() {
 		It("should createOrUpdate a new Kafka ACL", func() {
 			createdACL := &v1alpha1.KafkaACL{}
 			lookupKey := types.NamespacedName{Name: userName, Namespace: namespace}
-
 			Expect(k8sClient.Get(ctx, lookupKey, createdACL)).Should(Succeed())
 
-			By("by checking that after KafkaACL was created")
+			By("checking that after KafkaACL was created")
 			Expect(meta.IsStatusConditionTrue(createdACL.Status.Conditions, conditionTypeRunning)).Should(BeTrue())
+
+			By("checking initial permission before update")
+			Expect(createdACL.Spec.Permission).Should(Equal("admin"))
+
+			By(`updating permission from "admin" to "write" and making sure it has a new ID`)
+			createdACL.Spec.Permission = "write"
+			Expect(k8sClient.Update(ctx, createdACL)).Should(Succeed())
+
+			// It takes a time
+			var updatedACL *v1alpha1.KafkaACL
+			Eventually(func() bool {
+				updatedACL = &v1alpha1.KafkaACL{}
```

Comment:
It is not a good idea to query Aiven during the acc tests. In these tests, we are interested in observing K8s CR state transitions as our customers would. For example, once we altered the spec, a new generation was created, and the resource got reconciled and went through different conditions. From the customer's perspective, what should they do to get an updated ACL from their K8s application while waiting for that change?

Reply:
It queries k8s :)